tests/unit/data_formats/test_yolo_format.py

```diff
@@ -0,0 +1,1243 @@
+    AnnotationType,
+    Bbox,
+    LabelCategories,
+    Points,
+    PointsCategories,
+    Polygon,
+    Skeleton,
+)
+from datumaro.components.dataset import Dataset
+from datumaro.components.environment import Environment
+from datumaro.components.errors import (
+    AnnotationImportError,
+    DatasetExportError,
+    DatasetImportError,
+    DatasetNotFoundError,
+    InvalidAnnotationError,
+    ItemImportError,
+    UndeclaredLabelError,
+)
+from datumaro.components.extractor import DatasetItem
+from datumaro.components.format_detection import FormatDetectionContext, FormatRequirementsUnmet
+from datumaro.components.media import Image
+from datumaro.plugins.yolo_format.converter import (
+    Yolo8Converter,
+    Yolo8ObbConverter,
+    Yolo8PoseConverter,
+    Yolo8SegmentationConverter,
+    YoloConverter,
+)
+from datumaro.plugins.yolo_format.extractor import (
+    Yolo8Extractor,
+    Yolo8ObbExtractor,
+    Yolo8PoseExtractor,
+    Yolo8SegmentationExtractor,
+    YoloExtractor,
+)
+from datumaro.plugins.yolo_format.importer import (
+    Yolo8Importer,
+    Yolo8ObbImporter,
+    Yolo8PoseImporter,
+    Yolo8SegmentationImporter,
+    YoloImporter,
+)
+from datumaro.util.image import save_image
+from datumaro.util.test_utils import compare_datasets, compare_datasets_strict
+
+from ...requirements import Requirements, mark_requirement
+from ...utils.assets import get_test_asset_path
+
+
+def randint(a, b):
+    return random.randint(a, b)  # nosec B311
+
+
+class CompareDatasetMixin:
+    @pytest.fixture(autouse=True)
+    def setup(self, helper_tc):
+        self.helper_tc = helper_tc
+
+    def compare_datasets(self, expected, actual, **kwargs):
+        compare_datasets(self.helper_tc, expected, actual, **kwargs)
+
+
+class CompareDatasetsRotationMixin(CompareDatasetMixin):
+    def compare_datasets(self, expected, actual, **kwargs):
+        actual_copy = copy.deepcopy(actual)
+        compare_datasets(self.helper_tc, expected, actual, ignored_attrs=["rotation"], **kwargs)
+        for item_a, item_b in zip(expected, actual_copy):
+            for ann_a, ann_b in zip(item_a.annotations, item_b.annotations):
+                assert ("rotation" in ann_a.attributes) == ("rotation" in ann_b.attributes)
+                assert (
+                    abs(ann_a.attributes.get("rotation", 0) - ann_b.attributes.get("rotation", 0))
+                    < 0.01
+                )
+
+
+class YoloConverterTest(CompareDatasetMixin):
+    CONVERTER = YoloConverter
+    IMPORTER = YoloImporter
+
+    def _generate_random_bbox(self, n_of_labels=10, **kwargs):
+        return Bbox(
+            x=randint(0, 4),
+            y=randint(0, 4),
+            w=randint(1, 4),
+            h=randint(1, 4),
+            label=randint(0, n_of_labels - 1),
+            attributes=kwargs,
+        )
+
+    def _generate_random_annotation(self, n_of_labels=10):
+        return self._generate_random_bbox(n_of_labels=n_of_labels)
```

Comment:
It's considered a bad practice to have random tests, unless it's direct focus of the test. This makes them harder to reproduce and fail sporadically. Please consider moving to reproducible scenarios.